guest/prescore/prescore.go

```diff
@@ -81,46 +81,55 @@ func _prescore() uint32 {
 	// Pod is lazy and the same value for all plugins in a scheduling cycle.
 	pod := cyclestate.Pod
 
-	s := prescore.PreScore(cyclestate.Values, pod, &nodeList{})
+	s := prescore.PreScore(cyclestate.Values, pod, &filteredNodeInfoList{})
 
 	return imports.StatusToCode(s)
 }
 
-// nodeList is lazy so that a plugin which doesn't read fields avoids a
+// filteredNodeInfoList is lazy so that a plugin which doesn't read fields avoids a
 // relatively expensive unmarshal penalty.
 //
 // Note: Unlike proto.Pod, this is not special cased for the scheduling cycle.
-type nodeList struct {
-	items []proto.Node
+type filteredNodeInfoList struct {
+	// It only contains the Nodes that passed the filtering phase.
```

Comment:
Why do you define filteredNodeInfoList even though sharedLister has nodeinfos? Is this because wasm's prescore step is sometimes called without calling previous steps like prefilter?

Reply:
Because, as documented at this line, we have to know which nodes have gone thru the filtering phase.

As you can see in `filteredNodeInfoList.lazyItems`, actually `filteredNodeInfoList` utilizes nodes in `sharedLister` too to use the cached nodes there. 
Also, we don't want to put this "listing nodes after filtering" logic into `sharedLister` because we want to keep `sharedLister` pure, and "listing nodes that have gone thru the filtering phase" is specific to PreScore.